core/spec/models/spree/wallet/add_payment_sources_to_wallet_spec.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Spree::Wallet::AddPaymentSourcesToWallet, type: :model do
+  let(:order) { create(:order_ready_to_complete) }
+
+  describe '#add_to_wallet' do
+    subject { described_class.new(order) }
+
+    it 'saves the payment source and calls make_default' do
+      expect(subject).to receive(:make_default)
+      subject.add_to_wallet
+      expect(order.user.wallet.wallet_payment_sources.count).to eq(1)
+    end
```

Comment:
We do not need to explicitly test the implementation of `make_default` here, as you already have an expectation about the default payment source in the next test.

Also, I prefer to test changes like this:

```rb
expect { subject.add_to_wallet }.to change {
  order.user.wallet.wallet_payment_sources.count
}.by(1)
```